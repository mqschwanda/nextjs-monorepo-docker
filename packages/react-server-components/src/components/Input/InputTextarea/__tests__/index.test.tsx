import { render } from '@testing-library/react';
import { InputTextarea } from '..';

describe('@mqs/react-server-components', () => {
  describe('components', () => {
    describe('<InputTextarea />', () => {
      it('renders', () => {
        const testId = InputTextarea.name;

        const { getByTestId, asFragment } = render((
          <InputTextarea
            className='test'
            testId={testId}
          />
        ));

        const element = getByTestId(testId);
        expect(element).toBeTruthy();

        expect(asFragment()).toMatchSnapshot();
      });
    });
  });
});
