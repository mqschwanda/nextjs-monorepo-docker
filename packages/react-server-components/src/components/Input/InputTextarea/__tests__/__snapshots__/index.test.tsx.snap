// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@mqs/react-server-components components <InputTextarea /> renders 1`] = `
<DocumentFragment>
  <textarea
    class="textarea textarea-primary textarea-md test"
    data-testid="InputTextarea"
  />
</DocumentFragment>
`;

exports[`Storyshots @mqs/react-server-components/components/Input/InputTextarea Input Textarea Color Example 1`] = `
[
  <div
    className="form-control my-4"
    data-testid="FormControl"
  >
    <label
      className="label"
      data-testid="Label"
    >
      <span
        className="label-text"
      >
        accent
      </span>
    </label>
    <textarea
      className="textarea textarea-bordered textarea-accent textarea-md"
      data-testid="InputTextarea"
      placeholder="type here..."
    />
  </div>,
  <div
    className="form-control my-4"
    data-testid="FormControl"
  >
    <label
      className="label"
      data-testid="Label"
    >
      <span
        className="label-text"
      >
        error
      </span>
    </label>
    <textarea
      className="textarea textarea-bordered textarea-error textarea-md"
      data-testid="InputTextarea"
      placeholder="type here..."
    />
  </div>,
  <div
    className="form-control my-4"
    data-testid="FormControl"
  >
    <label
      className="label"
      data-testid="Label"
    >
      <span
        className="label-text"
      >
        ghost
      </span>
    </label>
    <textarea
      className="textarea textarea-bordered textarea-ghost textarea-md"
      data-testid="InputTextarea"
      placeholder="type here..."
    />
  </div>,
  <div
    className="form-control my-4"
    data-testid="FormControl"
  >
    <label
      className="label"
      data-testid="Label"
    >
      <span
        className="label-text"
      >
        info
      </span>
    </label>
    <textarea
      className="textarea textarea-bordered textarea-info textarea-md"
      data-testid="InputTextarea"
      placeholder="type here..."
    />
  </div>,
  <div
    className="form-control my-4"
    data-testid="FormControl"
  >
    <label
      className="label"
      data-testid="Label"
    >
      <span
        className="label-text"
      >
        primary
      </span>
    </label>
    <textarea
      className="textarea textarea-bordered textarea-primary textarea-md"
      data-testid="InputTextarea"
      placeholder="type here..."
    />
  </div>,
  <div
    className="form-control my-4"
    data-testid="FormControl"
  >
    <label
      className="label"
      data-testid="Label"
    >
      <span
        className="label-text"
      >
        secondary
      </span>
    </label>
    <textarea
      className="textarea textarea-bordered textarea-secondary textarea-md"
      data-testid="InputTextarea"
      placeholder="type here..."
    />
  </div>,
  <div
    className="form-control my-4"
    data-testid="FormControl"
  >
    <label
      className="label"
      data-testid="Label"
    >
      <span
        className="label-text"
      >
        success
      </span>
    </label>
    <textarea
      className="textarea textarea-bordered textarea-success textarea-md"
      data-testid="InputTextarea"
      placeholder="type here..."
    />
  </div>,
  <div
    className="form-control my-4"
    data-testid="FormControl"
  >
    <label
      className="label"
      data-testid="Label"
    >
      <span
        className="label-text"
      >
        warning
      </span>
    </label>
    <textarea
      className="textarea textarea-bordered textarea-warning textarea-md"
      data-testid="InputTextarea"
      placeholder="type here..."
    />
  </div>,
]
`;

exports[`Storyshots @mqs/react-server-components/components/Input/InputTextarea Input Textarea Example 1`] = `
<textarea
  className="textarea textarea-bordered textarea-primary textarea-md"
  data-testid="InputTextarea"
  placeholder="type here..."
/>
`;

exports[`Storyshots @mqs/react-server-components/components/Input/InputTextarea Input Textarea Size Example 1`] = `
[
  <div
    className="form-control my-4"
    data-testid="FormControl"
  >
    <label
      className="label"
      data-testid="Label"
    >
      <span
        className="label-text"
      >
        lg
      </span>
    </label>
    <textarea
      className="textarea textarea-bordered textarea-primary textarea-lg"
      data-testid="InputTextarea"
      placeholder="type here..."
    />
  </div>,
  <div
    className="form-control my-4"
    data-testid="FormControl"
  >
    <label
      className="label"
      data-testid="Label"
    >
      <span
        className="label-text"
      >
        md
      </span>
    </label>
    <textarea
      className="textarea textarea-bordered textarea-primary textarea-md"
      data-testid="InputTextarea"
      placeholder="type here..."
    />
  </div>,
  <div
    className="form-control my-4"
    data-testid="FormControl"
  >
    <label
      className="label"
      data-testid="Label"
    >
      <span
        className="label-text"
      >
        sm
      </span>
    </label>
    <textarea
      className="textarea textarea-bordered textarea-primary textarea-sm"
      data-testid="InputTextarea"
      placeholder="type here..."
    />
  </div>,
  <div
    className="form-control my-4"
    data-testid="FormControl"
  >
    <label
      className="label"
      data-testid="Label"
    >
      <span
        className="label-text"
      >
        xs
      </span>
    </label>
    <textarea
      className="textarea textarea-bordered textarea-primary textarea-xs"
      data-testid="InputTextarea"
      placeholder="type here..."
    />
  </div>,
]
`;
